core/shared/platform/common/posix/posix_socket.c

```diff
@@ -168,9 +176,16 @@ os_socket_bind(bh_socket_t socket, const char *host, int *port)
         goto fail;
     }
 
-    *port = ntohs(addr.ss_family == AF_INET
-                      ? ((struct sockaddr_in *)&addr)->sin_port
-                      : ((struct sockaddr_in6 *)&addr)->sin6_port);
+    if (addr.ss_family == AF_INET) {
+        *port = ntohs(((struct sockaddr_in *)&addr)->sin_port);
+    }
+    else {
+#ifdef IPPROTO_IPV6
```

Comment:
I'm not sure if we ever actually get to the `else` branch, I'd expect the `bind()` to fail sooner if IPv6 is not supported.

I think we can probably fail way before making any syscall if IPv6 is not supported; my suggestion would be to move textual_addr_to_sockaddr at the very top, and move this if...else just after that. What do you think?

Reply:
Makes sense, moved these bits to the start of this function.